### PR TITLE
Added 'openMVG2MVE2.cpp' file to translate from openMVG0.8-->MVE2 files

### PR DESCRIPTION
**Pull request by KevinCain**
This draft utility aims to convert openMVG0.8-->MVE2 files.

The MVE2 file output formatting seems fine as `dmrecon` is able to recognize and load the proper number of cameras and features, but `DMRecon::globalViewSelection()` finds no neighbors, reporting ‘Global View Selection failed’.

Sample call to `openMVG2MVE2` as follows:
./openMVG_main_openMVG2MVE2 -i ...openMVG_Build/software/SfM/tutorial_out/reconstruction_global/sfm_data.json -o /output_path

Sample call to `dmrecon`:
./dmrecon/dmrecon -s3 /output_path

For context please see this thread:  https://github.com/openMVG/openMVG/issues/325#issuecomment-123432556

### CODE REVIEW

File: src/software/SfM/main_openMVG2MVE2.cpp [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+#include <stdlib.h>
+#include <stdio.h>
+#include <cmath>
+#include <iterator>
+#include <iomanip>
+
+/* Notes:
+ * - An MVE2 scene appears to duplicate camera rot matrix and trans vector per-view data in 'meta.ini'
+ *   within the first section of 'synth_0.out'.
+ * - We do not save the original, instead we rely on the undistorted image from openMPV.
```
Comment (pmoulon) [inline src/software/SfM/main_openMVG2MVE2.cpp]: openMPV  => openMVG?

File: src/software/SfM/main_openMVG2MVE2.cpp [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+      os << std::setw(5) << std::setfill('0') << map_viewIdToContiguous[view->id_view];
+      std::string dstImage = stlplus::create_filespec(
+        stlplus::folder_append_separator(sOutViewIteratorDirectory), "undistorted","png");
+
+      const IntrinsicBase * cam = iterIntrinsic->second.get();
+      if (map_viewIdToContiguous[view->id_view] == 0)
+        w_h_image_size = std::make_pair(cam->w(), cam->h());
+      else
+      {
+        // check that there is no image sizing change -- do we need to enforce this for MVE?
```
Comment (pmoulon) [inline src/software/SfM/main_openMVG2MVE2.cpp]: @simonfuhrmann Can you confirm that MVE can support image with various size?

File: src/software/SfM/main_openMVG2MVE2.cpp [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+        }
+      }
+
+      // Prepare to write an MVE 'meta.ini' file for the current view
+      const Pose3 pose = sfm_data.GetPoseOrDie(view);
+      Mat34 P = cam->get_projective_equivalent(pose);
+
+      for ( int i = 1; i < 3 ; ++i)
+        for ( int j = 0; j < 4; ++j)
+          P(i, j) *= -1.;
```
Comment (pmoulon) [inline src/software/SfM/main_openMVG2MVE2.cpp]: The -1 should not be required.

File: src/software/SfM/main_openMVG2MVE2.cpp [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+      Mat3 R, K;
+      Vec3 t;
+      KRt_From_P( P, &K, &R, &t);
+
+      // Output translation via optical center vector for given pose
+      const Vec3 optical_center = R.transpose() * t;
```
Comment (pmoulon) [inline src/software/SfM/main_openMVG2MVE2.cpp]: Use directly pose.center()

File: src/software/SfM/main_openMVG2MVE2.cpp [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+      // Prepare to write an MVE 'meta.ini' file for the current view
+      const Pose3 pose = sfm_data.GetPoseOrDie(view);
+      Mat34 P = cam->get_projective_equivalent(pose);
+
+      for ( int i = 1; i < 3 ; ++i)
+        for ( int j = 0; j < 4; ++j)
+          P(i, j) *= -1.;
+
+      Mat3 R, K;
```
Comment (pmoulon) [inline src/software/SfM/main_openMVG2MVE2.cpp]: R = pose.R()

File: src/software/SfM/main_openMVG2MVE2.cpp [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+
+      // Output translation via optical center vector for given pose
+      const Vec3 optical_center = R.transpose() * t;
+      // Pixel aspect = pixel width divided by the pixel height
+      const float pixelAspect = cam->w()/cam->h();
+      // Focal length and principal point are embedded into the calibration matrix K:
+      // focal_length = K(0,0)
+      // const std::vector<double> pp = {_K(0,2), _K(1,2)}
+      // their values are normalized (0..1)
+      const float flen = cam->h()/K(0, 0);
```
Comment (pmoulon) [inline src/software/SfM/main_openMVG2MVE2.cpp]: According https://github.com/simonfuhrmann/mve/commit/12ca8538bd735b28b3a4da9e22da31b09cea9191 flen must be equal to cam->focal() / std::max(cam->h(), cam->w() );

**Comment by pmoulon** [comment]
Thanks @KevinCain 
I have made some comment in the commit that could explain the reason why MVE report a fail in DMRecon::globalViewSelection() => "finds no neighbors, reporting ‘Global View Selection failed’."

**Comment by KevinCain** [comment]
Thanks for these good points!  Once @simonfuhrmann has to a chance to weigh in on the MVE-specific questions I'll address these and continue testing.

**Comment by KevinCain** [comment]
Thanks for all the fast, specific feedback.  I've incorporated your notes with my latest edits and am now able to get more sensible results.

- I agree that pixel aspect will normally (almost always, in fact) be 1.  However I noted that in openMVG output, cam->w() / cam->h() = 1.  Why is this?  That implies a square aspect ratio, which the testing source images do not have.  
- I believe the view IDs are correct but would appreciate notes.
- I haven't removed view sorting yet but will as testing proceeds.
- I agree we want to add thumbnails but can add this once the core conversion is working correctly.

**Comment by pmoulon** [comment]
Thanks guys (@KevinCain @simonfuhrmann)
First draft have been accepted with minor modifications.
Will add a new issue for the Thumbnail generation.

**Pull request merged**